### ISSUE TITLE
Low level documentation

### ISSUE THREAD

**Issue by jbltx** [stride3d/stride-docs#39]
As discussed on Discord server, it seems there's no real low level documentation about the core concepts of the Engine. It should be really useful to document them in order to contribute at the project, or at least as reference. For example concepts regarding the **Build system** and the **Serialization of data** in general. And the Big-Picture of Stride Engine (as mentionned @xen2 ).

Some questions in my mind :
- What is the difference between a ContentSerializer and a `DataSerializer` ? 
- How is composed a chunk of serialized data ? 
- How the `IReference` interface works together with the serialization system ? 

I am of course reading comments on all classes but I think there are some missing notions which seem to be important for me.

As another user mentionned on Discord, a readme file per csproj should be already a great start.

Thank you.

**Comment by Kryptos-FR** [comment]
There are a few old documents here: https://github.com/stride3d/stride/tree/master/docs/technical
But it is indeed a bit sparse.

**Comment by Jklawreszuk** [comment]
I agree, I had similar concept recently in mind (esp. how engine works, general workflow etc). 
We could integrate technical documentation into separate category or the same category as manual. 
It would contain basic information about the engine for potential contributors, as well as unusual solutions that may not be understood without the help of someone experienced with an engine.

**Comment by VaclavElias** [comment]
@Aggror already started with the new section for the contributors, which will be part of Stride Docs

https://github.com/stride3d/stride-docs/tree/contributor-section%2Bwiki/en/contributors

Once ready, we can further shape and update accordingly and you can help also with the content :)

**Comment by VaclavElias** [comment]
Just a follow up. Some content is published but lots of more need to be done..

https://doc.stride3d.net/latest/en/contributors/engine/index.html
